controllers/apim/utils.go

```diff
@@ -0,0 +1,81 @@
+package apim
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/kuadrant/limitador-operator/api/v1alpha1"
+	istio "istio.io/client-go/pkg/apis/networking/v1alpha3"
+	istiosecurityv1beta1 "istio.io/client-go/pkg/apis/security/v1beta1"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// gatewayLabels fetches labels of an Istio gateway identified using the given ObjectKey.
+func gatewayLabels(ctx context.Context, client client.Client, gwKey client.ObjectKey) map[string]string {
+	gateway := &istio.Gateway{}
+	if err := client.Get(ctx, gwKey, gateway); err != nil {
+		return map[string]string{}
+	}
+	return gateway.Labels
+}
+
+// rlFiltersPatchName returns the name of EnvoyFilter adding in rate-limit filters to a gateway.
+func rlFiltersPatchName(gatewayName string) string {
+	return gatewayName + "-ratelimit-filters"
+}
+
+// ratelimitsPatchName returns the name of EnvoyFilter adding in ratelimits to a gateway.
+func ratelimitsPatchName(rlpName, gatewayName string) string {
+	return fmt.Sprintf("ratelimits-of-%s-in-%s", rlpName, gatewayName)
```

Comment:
The envoy filter contains `Host` which belongs to the VS. So I wonder if the EnvoyFitler Name should be indexed by the VS as well.